### PR TITLE
feat(native): project foldable controls with SwiftUI

### PR DESCRIPTION
**Pull request by rdlabo**
## Summary

- hand foldable tabs, back navigation, menu buttons, and eligible fixed-toolbar actions from the Web fallback to a native SwiftUI adaptive surface when iOS exposes a physical side rail
- keep Ionic controls as the source of truth and forward native activation through the existing event, routing, and form-submit path
- clip and hit-test the hosting surface to the physical Duo rail so Web content remains visible and interactive elsewhere
- keep `ion-menu`, `ion-modal`, and `ion-popover` as independent surfaces
- retain the Web projection on Web, Android, older iOS, older plugin versions, and supported iOS devices without a foldable rail

## State and ownership

- `placement: foldable-rail` is an optional, backward-compatible wire hint
- SwiftUI owns adaptive placement and Liquid Glass; Ionic owns labels, icons, disabled/selected state, actions, and navigation
- tab taps update optimistically, survive stale Web snapshots, confirm on matching Web state, and roll back on timeout, disablement, or target removal
- `ion-buttons` remains one native group; opted-out groups continue to project eligible child buttons independently
- Native and Web fallback handles report their combined ownership through the public status contract

## Stack

1. #213 — Web-first foldable layout and controls
2. **This PR — Native-first SwiftUI ownership for controls relocated into the foldable rail**

Base: `feat/foldable-toggle`

## Verification

- root TypeScript/CSS build and lint
- Swift package tests on iPhone Duo Simulator: 23 passed
- Native UI Shell Playwright suite: 69 passed without exclusions
- iPhone Duo iOS 27.1 UI test: native tabs, selected state, menu, submit, and back navigation passed
- Xcode iPhone Duo build passed

## Review

- self-review: approved
- senior review cycle: approved after clarifying ownership-boundary coverage
- independent acceptance review cycle: approved after adding the non-Duo capability fallback and combined status accounting
- OSS review: approved after documenting native ownership versus Web fallback

### CODE REVIEW

File: ios/Sources/IonicNativeUIShellPlugin/Components/ShellFoldableRail.swift [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+@available(iOS 26.0, *)
+final class ShellFoldableRailModel: ObservableObject {
+    struct Item: Identifiable {
+        let id: String
+        let label: String
+        let accessibilityLabel: String
+        let image: UIImage?
+        let disabled: Bool
+        var selected: Bool
+    }
```
[the remark this answers is not in the document]
Comment (rdlabo) [reply ios/Sources/IonicNativeUIShellPlugin/Components/ShellFoldableRail.swift:22]: 修正しました。既存 `ShellBadge` をrail modelへ保持し、SwiftUI TabView itemの標準 `.badge()` に渡します。空文字notification badgeもoptionalとして保持し、model testも追加しました。

File: ios/Sources/IonicNativeUIShellPlugin/IonicNativeUIShellPlugin.swift [hunk 2]
```diff
@@ -216,6 +223,18 @@ public class IonicNativeUIShellPlugin: CAPPlugin, CAPBridgedPlugin, UITabBarDele
...
             var rejectedSearches: [String] = []
+            if foldable.isEmpty {
+                self.foldableRail?.detach()
+                self.foldableRail = nil
+            } else if let owner = self.bridge?.viewController {
+                let rail = self.foldableRail ?? ShellFoldableRailController(activate: { [weak self] id in self?.activate(id) })
+                self.foldableRail = rail
+                rail.attach(to: owner, in: owner.view)
+                rail.apply(foldable, rendering: self.rendering)
+                rail.view.isHidden = false
```
[the remark this answers is not in the document]
Comment (rdlabo) [reply ios/Sources/IonicNativeUIShellPlugin/IonicNativeUIShellPlugin.swift]: 修正しました。`keyboardVisible` 中はrailをdetachし、全foldable control IDを `rejectedControls` として返して既存のWeb ownership handoffを使います。keyboard hide後は既存refreshで再評価されます。

File: src/native/components/index.ts [hunk 3]
```diff
@@ -27,13 +27,22 @@ export const motionSelector = [
   ...components.filter((component) => 'tracksMotion' in component && component.tracksMotion).map((component) => component.tag),
 ].join(', ');
 
+const foldableRailTags = new Set(['ion-button', 'ion-back-button', 'ion-buttons', 'ion-menu-button', 'ion-tab-bar']);
+
+export const isFoldableRailCandidate = (element: HTMLElement): boolean =>
+  foldableRailTags.has(element.localName) &&
+  (!element.matches('ion-button') || !element.parentElement?.matches('ion-buttons') || isDisabledButtonGroupChild(element)) &&
+  !element.closest('ion-menu, ion-modal, ion-popover') &&
+  !!element.closest(':is(ion-app, body).ios-theme-enable-foldable');
```
[the remark this answers is not in the document]
Comment (rdlabo) [reply src/native/components/index.ts]: 修正しました。単一の通常 `ion-button` だけは child candidate として `foldable-rail` を受け取り、複数buttonは引き続き親groupだけが所有します。既存foldable E2Eにsingle-actionへの遷移assertionも追加しました。

**Comment by rdlabo** [comment]
Review cycle completed.

- Self-review: confirmed the tests are split by contract (DOM/native ownership, Swift state reconciliation, and real Duo interaction) without duplicating route screenshot coverage.
- Senior review: requested clearer ancestor/element opt-out and ordinary-fill ownership boundaries; fixed and re-reviewed to approval.
- Independent acceptance review: found the missing non-Duo capability boundary and combined status accounting. Added an explicit native `foldableRail` capability, retained Web projection when unavailable, preserved ordinary native controls, and fixed the public status aggregation. Re-reviewed to approval with no remaining findings.
- Validation: lint and TypeScript/CSS build passed; Native UI Shell Playwright suite 69/69 passed; Swift package tests on iPhone Duo Simulator 23/23 passed.

The fallback test verifies native non-ownership, visible/operable Web back and tab controls, and the public projected-count contract. Commit: 69bdf05c.
